XamarinCommunityToolkit/Views/CameraView/Android/CameraFragment.android.cs

```diff
@@ -0,0 +1,960 @@
+			mediaRecorder.SetPreviewDisplay(previewSurface);
+			if (audioPermissionsGranted)
+				mediaRecorder.SetAudioSource(AudioSource.Camcorder);
+			mediaRecorder.SetVideoSource(AVideoSource.Surface);
+
+			var profile = GetCamcoderProfile();
+			if (profile != null)
+			{
+				mediaRecorder.SetProfile(profile);
+			}
+			else
+			{
+				mediaRecorder.SetOutputFormat(OutputFormat.Mpeg4);
+				mediaRecorder.SetVideoEncodingBitRate(10000000);
+				mediaRecorder.SetVideoFrameRate(30);
+				mediaRecorder.SetVideoSize(videoSize.Width, videoSize.Height);
+				mediaRecorder.SetVideoEncoder(VideoEncoder.H264);
+				if (audioPermissionsGranted)
+				{
+					mediaRecorder.SetAudioEncoder(AudioEncoder.Default);
+				}
+			}
+
+			videoFile = ConstructMediaFilename("VID", "mp4");
+
+			mediaRecorder.SetOutputFile(videoFile);
+			mediaRecorder.SetOrientationHint(GetCaptureOrientation());
+			mediaRecorder.Prepare();
+		}
+
+		CamcorderProfile GetCamcoderProfile()
+		{
+			var cameraId = Convert.ToInt32(this.cameraId);
+			if (CamcorderProfile.HasProfile(cameraId, CamcorderQuality.HighSpeed1080p))
+				return CamcorderProfile.Get(cameraId, CamcorderQuality.HighSpeed1080p);
+			else if (CamcorderProfile.HasProfile(cameraId, CamcorderQuality.HighSpeed720p))
+				return CamcorderProfile.Get(cameraId, CamcorderQuality.HighSpeed720p);
+			else if (CamcorderProfile.HasProfile(cameraId, CamcorderQuality.HighSpeed480p))
+				return CamcorderProfile.Get(cameraId, CamcorderQuality.HighSpeed480p);
+			else if (CamcorderProfile.HasProfile(cameraId, CamcorderQuality.High))
+				return CamcorderProfile.Get(cameraId, CamcorderQuality.High);
+			else if (CamcorderProfile.HasProfile(cameraId, CamcorderQuality.Low))
+				return CamcorderProfile.Get(cameraId, CamcorderQuality.Low);
+
+			return null;
+		}
+
+		public void StartRecord()
+		{
+			if (IsBusy)
+			{
+				return;
+			}
+			else if (IsRecordingVideo)
+			{
+				Element?.RaiseMediaCaptureFailed("Video already recording.");
+				return;
+			}
+			else if (cameraTemplate != CameraTemplate.Record)
+			{
+				Element?.RaiseMediaCaptureFailed($"Unexpected error: Camera {cameraTemplate} not configured to record video.");
+				return;
+			}
+			else if (mediaRecorder == null)
+			{
+				Element?.RaiseMediaCaptureFailed($"Unexpected error: MediaRecorder is not initialized.");
+				IsRecordingVideo = false;
+				return;
+			}
+
+			try
+			{
+				Sound(MediaActionSoundType.StartVideoRecording);
+				mediaRecorder.Start();
+				IsRecordingVideo = true;
+			}
+			catch (Java.Lang.Exception error)
+			{
+				LogError("Failed to take video", error);
+				Element?.RaiseMediaCaptureFailed($"Failed to take video: {error}");
+				DisposeMediaRecorder();
+			}
+		}
+
+		public async void StopRecord()
+		{
+			if (IsBusy || !IsRecordingVideo || session == null || mediaRecorder == null)
+				return;
+
+			try
+			{
+				DisposeMediaRecorder();
+				await PrepareSession();
+			}
+			catch (Java.Lang.Exception ex)
+			{
```

Comment:
brackets, I'll stop now :)

Reply:
this one is needed, no?

Reply:
Yeah, my bad.  Forgot you can't for catch blocks.

Reply:
No problem